R/use_extendr.R

```diff
@@ -55,10 +51,10 @@ void R_init_{pkg_name}(void *dll) {{
 }}
 )"
   )
-  brio::write_lines(entrypoint_content, file.path(src_dir, "entrypoint.c"))
+  usethis::write_over(fs::path(src_dir, "entrypoint.c"), entrypoint_content, quiet = quiet)
```

Comment:
Is there any good reason to use `usethis::write_over()` rather than `brio::write_lines()`. It's the main reason we need the {usethis} dependency, I believe.

Reply:
There are several key differences between `brio::write_lines()` and `usethis::write_over()`:
1. `write_over()` informs the user if such file already exists. The user is asked whether to replace the file or not.
I find it useful in `use_extendr`, where we write a bunch of files. Imagine that the user already created `Makevars`, and we silently overwrite it. `write_over()` can help us avoid this.
2. `write_over()` output nice, styled message, something like `Writting to {file}`, which can be suppressed using `quiet` argument. with `write_lines()` we have to have an additional `if (!isTRUE(quiet)) { cli::cli_alert_success("Writing to {.file path}")}`
3. `write_over()` is useless for wrapper generation, because we need to have the most recent wrappers and can safely overwrite the existing file.

If {usethis} stays as package dependency, I think `write_over()` can be used for package set up. If not, we can either overwrite everything (not sure this is a great user experience) or make our own function (again, a few lines of code, nothing fancy or complicated).

Reply:
> Imagine that the user already created `Makevars`

I believe in my first draft of all of this I wasn't doing anything if `src` was already in place. Is this check still there or has it been removed?

Reply:
You are correct, and it is still there. If you would like to keep the original behavior, writing only if `src/` is not present, then we can switch back to {brio}.

Reply:
I think that's a save approach for now. Check if `src` exists already, and if it does, don't do anything. It means we can make much stronger assumptions throughout the rest of the codebase.